### ISSUE TITLE
Debugger improvements

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1184]
_From [Tsur.Her...@gmail.com](https://code.google.com/u/105782936570157446535/) on 2012-11-04T07:55:45Z_

This is a feature request / improvement suggestion;

1.make the breakpoint colour red big and noticeable .. I'll settle for just red
2. right click on breakpoint enables adding condition to it
3. bind f9/f10/f11 when gdb runs for step/continue/step into 
same like matlab.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1184_

**Comment by spyder-issues-migrator** [comment]
_From [Tsur.Her...@gmail.com](https://code.google.com/u/105782936570157446535/) on 2012-11-04T05:05:48Z_

I forgot to mention ... Spyder is an excellent and impressive tool

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-02-13T09:00:04Z_

We'll consider your suggestions for the next version. Thanks for your interest in Spyder.

I have another suggestion myself: change the color of the currently selected line from the normal one to a reddish one, to inform the user he's in debugging mode.

**Summary:** Debugger improvements  \<span class="oldvalue"> (was: Debugger Integrartion and breakpoint color) \</span>  
  **Labels:** MS-v2.3 Cat-Debugger

**Comment by spyder-issues-migrator** [comment]
_From [sylvain....@gmail.com](https://code.google.com/u/102212202545902308399/) on 2013-04-19T14:34:37Z_

It would be nice to draw an arrow in the margin at the position of the current line in execution during debugging. 

Although, it will probably be complicated to remove/redraw it when the focus goes on a different python console. I wonder if it is possible to ask an ipython console if whether pdb is launched and what is the current line...

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-05-11T06:00:29Z_

**Labels:** -MS-v2.3 MS-v2.4

**Comment by CAM-Gerlach** [comment]
> 1.make the breakpoint colour red big and noticeable .. I'll settle for just red

Its red now and almost as big as it can be, but will be made much easier to see and interact with #5283 and the associated #5005

> 2. right click on breakpoint enables adding condition to it

This is the only real remaining novel feature that I can tell. @dalthviz you want to add this as part of #5283 or a follow-on PR?

> 3. bind f9/f10/f11 when gdb runs for step/continue/step into

Already possible by simply rebinding the shortcuts, and a matlab etc shortcuts profile will be added in Spyder 4 which is covered elsewhere.

**Comment by dalthviz** [comment]
@CAM-Gerlach I'm quite not sure, @ccordoba12 what do you think?

**Comment by ccordoba12** [comment]
Sounds good, but this should be done for Spyder 4, not 3.

**Comment by CAM-Gerlach** [comment]
Yes, of course, since it would be part of the new debugger panel UI.

**Comment by ccordoba12** [comment]
Yep, that's right.
